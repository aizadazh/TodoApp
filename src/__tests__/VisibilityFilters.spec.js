import reducer from "../redux/reducers/visibilityFilter";
import { VISIBILITY_FILTERS } from "../constants";
import { SET_FILTER } from "../redux/actionTypes";


describe('Testing VisibilityFilters component', () => {
   it('should return an initial state', () => {
        expect(reducer(undefined, {})).toEqual(VISIBILITY_FILTERS.ALL)
   })
   it('should handle Set_Filter action', () => {
        const completedState = VISIBILITY_FILTERS.COMPLETED;
        const action = {
            type: SET_FILTER,
            payload: {
                filter: 'completed'
            }            
        }
          expect(reducer(completedState, action)).toEqual(VISIBILITY_FILTERS.COMPLETED)
          
    })
})
           
                