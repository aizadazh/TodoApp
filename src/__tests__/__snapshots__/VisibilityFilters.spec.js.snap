// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing VisibilityFilters component renders correctly 1`] = `ShallowWrapper {}`;
